.github/workflows/_build.yml

```diff
@@ -256,3 +256,107 @@ jobs:
+      contents: write
+    timeout-minutes: 30
+
+    strategy:
+      fail-fast: true
+
+    steps:
+
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Set up JDK 11
+        uses: actions/setup-java@v3
+        with:
+          java-version: '11'
+          distribution: 'adopt'
+          cache: gradle
+
+      - name: AVD cache
+        uses: actions/cache@v3
+        id: avd-cache
+        with:
+          path: |
+            ~/.android/avd/*
+            ~/.android/adb*
+          key: avd-${{ env.api-level }}
+
+      - name: Delete existing screenshots
+        run: |
+          rm -rf docs/view/*/screenshots/*
+          rm -rf docs/compose/*/screenshots/*
+
+      - name: create AVD and generate snapshot for caching
+        if: steps.avd-cache.outputs.cache-hit != 'true'
+        uses: reactivecircus/android-emulator-runner@v2
+        with:
+          profile: Pixel
+          sdcard-path-or-size: 512M
+          api-level: 31
+          force-avd-creation: false
+          emulator-options: -no-window -gpu swiftshader_indirect -noaudio -no-boot-anim -camera-back none
+          disable-animations: false
+          script: echo "Generated AVD snapshot for caching."
+
+      - name: Screenshots recording
+        id: screenshotsRecording
+        uses: reactivecircus/android-emulator-runner@v2
+        with:
+          profile: Pixel
+          sdcard-path-or-size: 512M
+          api-level: 31
+          force-avd-creation: false
+          emulator-options: -no-snapshot-save -no-window -gpu swiftshader_indirect -noaudio -no-boot-anim -camera-back none
+          disable-animations: true
+          script: |
+            touch logcat_screenshots.log
+            chmod 777 logcat_screenshots.log
+            adb logcat > logcat_screenshots.log &
+            ./scripts/generate_screenshots.sh
+            changedFiles=`git status --porcelain` && echo "CHANGED_FILES=${changedFiles//$'\n'/'%0A'}" >> $GITHUB_OUTPUT
+
+      - name: Upload Logcat Logs
+        if: ${{ failure() }}
+        uses: actions/upload-artifact@v3
+        with:
+          name: ${{ env.flavour }}${{ env.config }}-${{ matrix.variant }}-logcat-output
+          path: logcat_screenshots.log
+          retention-days: 5
+
+      - name: Notify PR of screenshot changes
+        if: ${{ (failure() || needs.CheckCommitMessage.outputs.head-commit-message == 'Record screenshots') && steps.screenshotsRecording.outputs.CHANGED_FILES != '' }}
+        uses: actions/github-script@v6
+        with:
+          script: |
+            var body = `### Screenshots were updated. Please verify the changes match the expected output. \n\n>'${{ steps.screenshotsRecording.outputs.CHANGED_FILES }}'`;
+
+            github.rest.issues.createComment({
+              issue_number: context.issue.number,
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              body: body
+            })
+
+      - name: Commit changes
```

Comment:
right now our docs screenshots aren't 100% reproducable - I believe the (old) calendar one changes depending on the date still. we need to fix that first before we enable this otherwise it will always pollute our PRs